super::clone breaks SuperCloneCheck

https://checkstyle.org/config_coding.html#SuperClone
https://checkstyle.org/config_coding.html#SuperFinalize

```
/var/tmp $ javac --release 11 A.java
# This only works in Java 11+

/var/tmp $ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">

<module name="Checker">
  <property name="charset" value="UTF-8" />

  <module name="TreeWalker">
    <module name="SuperCloneCheck" />
  </module>
</module>


/var/tmp $ cat A.java
class A {
  @FunctionalInterface
  public interface CheckedSupplier<R, E extends Exception> {
    R get() throws E;
  }

  public Object clone() {
    CheckedSupplier<Object, CloneNotSupportedException> r = super::clone; // NPE consuming this
    try {
      return r.get();
    } catch (CloneNotSupportedException e) {
      throw new RuntimeException(e);
    }
  }
}
 
/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-8.34-all.jar -c config.xml YOUR_FILE.java
Starting audit...
com.puppycrawl.tools.checkstyle.api.CheckstyleException: Exception was thrown while processing A.java
	at com.puppycrawl.tools.checkstyle.Checker.processFiles(Checker.java:311)
	at com.puppycrawl.tools.checkstyle.Checker.process(Checker.java:221)
	at com.puppycrawl.tools.checkstyle.Main.runCheckstyle(Main.java:408)
	at com.puppycrawl.tools.checkstyle.Main.runCli(Main.java:331)
	at com.puppycrawl.tools.checkstyle.Main.execute(Main.java:190)
	at com.puppycrawl.tools.checkstyle.Main.main(Main.java:125)
Caused by: java.lang.NullPointerException
	at com.puppycrawl.tools.checkstyle.checks.coding.AbstractSuperCheck.hasArguments(AbstractSuperCheck.java:142)
	at com.puppycrawl.tools.checkstyle.checks.coding.AbstractSuperCheck.isSuperCall(AbstractSuperCheck.java:106)
	at com.puppycrawl.tools.checkstyle.checks.coding.AbstractSuperCheck.visitToken(AbstractSuperCheck.java:85)
	at com.puppycrawl.tools.checkstyle.TreeWalker.notifyVisit(TreeWalker.java:340)
	at com.puppycrawl.tools.checkstyle.TreeWalker.processIter(TreeWalker.java:451)
	at com.puppycrawl.tools.checkstyle.TreeWalker.walk(TreeWalker.java:278)
	at com.puppycrawl.tools.checkstyle.TreeWalker.processFiltered(TreeWalker.java:151)
	at com.puppycrawl.tools.checkstyle.api.AbstractFileSetCheck.process(AbstractFileSetCheck.java:87)
	at com.puppycrawl.tools.checkstyle.Checker.processFile(Checker.java:337)
	at com.puppycrawl.tools.checkstyle.Checker.processFiles(Checker.java:298)
	... 5 more
Checkstyle ends with 1 errors.

```

---------------

**Describe what you expect in detail.**
I'm unsure if the built in check should pass when you make a reference to the superclass implementation. I was looking at extending `AbstractSuperCheck` to force calling `super.close()` in implementations of `close`. In my case it is normal and common to do stuff like `IOUtils.close(super::close, other, stuff, here);`. So, at least for me, it'd be best if `AbstractSuperCheck` considered a reference to the superclass implementation to pass. Or if it could be configured to do so.

--------------

--- a/src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/AbstractSuperCheck.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/AbstractSuperCheck.java
@@ -98,13 +98,12 @@ public abstract class AbstractSuperCheck
     private boolean isSuperCall(DetailAST literalSuperAst) {
         boolean superCall = false;
 
-        if (literalSuperAst.getType() == TokenTypes.LITERAL_SUPER) {
-            // dot operator?
-            final DetailAST dotAst = literalSuperAst.getParent();
-
-            if (!isSameNameMethod(literalSuperAst)
-                && !hasArguments(dotAst)) {
-                superCall = isSuperCallInOverridingMethod(dotAst);
+        if (literalSuperAst.getType() == TokenTypes.LITERAL_SUPER
+            && !isSameNameMethod(literalSuperAst)) {
+            final DetailAST parent = literalSuperAst.getParent();
+            if (parent.getType() == TokenTypes.METHOD_REF
+                || !hasArguments(parent)) {
+                superCall = isSuperCallInOverridingMethod(parent);
             }
         }
         return superCall;

--- a/src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/SuperCloneCheckTest.java
+++ b/src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/SuperCloneCheckTest.java
@@ -36,6 +36,7 @@ import com.puppycrawl.tools.checkstyle.JavaParser;
 import com.puppycrawl.tools.checkstyle.api.DetailAST;
 import com.puppycrawl.tools.checkstyle.api.TokenTypes;
 import com.puppycrawl.tools.checkstyle.internal.utils.TestUtil;
+import com.puppycrawl.tools.checkstyle.utils.CommonUtil;
 
 public class SuperCloneCheckTest
     extends AbstractModuleTestSupport {
@@ -66,6 +67,14 @@ public class SuperCloneCheckTest
             "9:17: " + getCheckMessage(MSG_KEY, "clone", "super.clone"),
         };
         verify(checkConfig, getPath("InputSuperClonePlainAndSubclasses.java"), expected);
+    }
+
+    @Test
+    public void testMethodReference() throws Exception {
+        final DefaultConfiguration checkConfig =
+            createModuleConfig(SuperCloneCheck.class);
+        final String[] expected = CommonUtil.EMPTY_STRING_ARRAY;
+        verify(checkConfig, getPath("InputSuperCloneMethodReference.java"), expected);
     }
 
     @Test

--- a/src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/SuperFinalizeCheckTest.java
+++ b/src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/SuperFinalizeCheckTest.java
@@ -39,10 +39,21 @@ public class SuperFinalizeCheckTest
         final DefaultConfiguration checkConfig =
             createModuleConfig(SuperFinalizeCheck.class);
         final String[] expected = {
-            "27:17: " + getCheckMessage(MSG_KEY, "finalize", "super.finalize"),
-            "34:17: " + getCheckMessage(MSG_KEY, "finalize", "super.finalize"),
+            "29:17: " + getCheckMessage(MSG_KEY, "finalize", "super.finalize"),
+            "36:17: " + getCheckMessage(MSG_KEY, "finalize", "super.finalize"),
+            "78:20: " + getCheckMessage(MSG_KEY, "finalize", "super.finalize"),
         };
         verify(checkConfig, getPath("InputSuperFinalizeVariations.java"), expected);
+    }
+
+    @Test
+    public void testMethodReference() throws Exception {
+        final DefaultConfiguration checkConfig =
+            createModuleConfig(SuperFinalizeCheck.class);
+        final String[] expected = {
+            "18:20: " + getCheckMessage(MSG_KEY, "finalize", "super.finalize"),
+        };
+        verify(checkConfig, getPath("InputSuperFinalizeMethodReference.java"), expected);
     }
 
 }

--- a/src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/superclone/InputSuperCloneMethodReference.java
+++ b/src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/superclone/InputSuperCloneMethodReference.java
@@ -1,0 +1,21 @@
+package com.puppycrawl.tools.checkstyle.checks.coding.superclone;
+
+import java.util.ArrayList;
+
+/** Config = default */
+public class InputSuperCloneMethodReference extends ArrayList {
+
+    public interface CheckedSupplier<R, E extends Exception> {
+        R get() throws E;
+    }
+
+    public Object clone() { // ok
+        CheckedSupplier<Object, CloneNotSupportedException> r = super::clone;
+        CheckedSupplier<Object, CloneNotSupportedException> r1 = super::getClass;
+        try {
+            return r.get();
+        } catch (CloneNotSupportedException e) {
+            return null;
+        }
+    }
+}

--- a/src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/superfinalize/InputSuperFinalizeMethodReference.java
+++ b/src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/superfinalize/InputSuperFinalizeMethodReference.java
@@ -1,0 +1,20 @@
+package com.puppycrawl.tools.checkstyle.checks.coding.superfinalize;
+
+/** Config = default */
+public class InputSuperFinalizeMethodReference extends ClassWithFinalizer {
+
+    public interface CheckedConsumer<E extends Throwable> {
+        void get() throws E;
+    }
+
+    @Override
+    protected void finalize() throws Throwable {
+        CheckedConsumer<Throwable> r = super::finalize;
+        r.get();
+    }
+}
+
+class ClassWithFinalizer {
+    protected void finalize() throws Throwable { // violation
+    }
+}

--- a/src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/superfinalize/InputSuperFinalizeVariations.java
+++ b/src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/superfinalize/InputSuperFinalizeVariations.java
@@ -1,4 +1,6 @@
 package com.puppycrawl.tools.checkstyle.checks.coding.superfinalize;
+
+/** Config = default */
 public class InputSuperFinalizeVariations
 {
     public InputSuperFinalizeVariations() throws Throwable
@@ -24,14 +26,14 @@ public class InputSuperFinalizeVariations
 
 class NoSuperFinalize
 {
-    public void finalize()
+    public void finalize() // violation
     {
     }
 }
 
 class InputInnerFinalize
 {
-    public void finalize()
+    public void finalize() // violation
     {
         class Inner
         {
@@ -64,5 +66,16 @@ class InputOneMore {
                 super.finalize();
             }
         }
+    }
+}
+
+class FinalizeWithArgs {
+    public void finalize(Object a) {};
+}
+
+class OverrideClass extends FinalizeWithArgs {
+    @Override
+    protected void finalize() throws Throwable { // violation
+        super.finalize(new Object());
     }
 }

